src/NumericalAlgorithms/RootFinding/GslMultiRoot.cpp

```diff
@@ -0,0 +1,151 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "GslMultiRoot.hpp"
+
+#include <exception>
+#include <gsl/gsl_multiroots.h>
+#include <gsl/gsl_vector.h>
+
+#include "ErrorHandling/Assert.hpp"
+#include "ErrorHandling/Error.hpp"
+#include "Parallel/Printf.hpp"
+
+class rootfinding_exception : public std::exception {
+  virtual const char* what() const throw() { return "GSLMultiRoot failed."; }
+};
+
+void printstate_GSLMultiRootFinder(size_t iter, gsl_multiroot_fdfsolver* s,
+                                   size_t dim, bool PrintHeader = false) {
```

Comment:
`print_header`
Make dim a template parameter.
Currently this is only called with PrintHeader true.  Is there an expected use case for the false version?

Reply:
I haven't thought of one, but I can make it so that the false version is called when a particular Verbosity is set